module/VuFind/src/VuFind/Db/Service/SearchService.php

```diff
@@ -161,22 +170,37 @@ public function getSearchByIdAndOwner(
      */
     public function getSearches(?string $sessionId, UserEntityInterface|int|null $userOrId = null): array
     {
-        // If we don't get a session id or user id, don't return anything:
-        if (null === $sessionId && null === $userOrId) {
+        $userId = $userOrId instanceof UserEntityInterface ? $userOrId->getId() : $userOrId;
+
+        if ($sessionId == null && $userId == null) {
```

Comment:
Might be worth using strict comparison:

```suggestion
        if ($sessionId === null && $userId === null) {
```